File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/02 Print name N times/printName.java
// https://www.geeksforgeeks.org/problems/print-gfg-n-times/1

class Solution {

    public void printName(String name, int n) {
        // Base case to stop recursion
        if (n == 0)
            return;
        System.out.println(name);
        printName(name, n - 1);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        solution.printName("Shreyans", 5);
    }
}

File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/02 Print name N times/printName.py
# https://www.geeksforgeeks.org/problems/print-gfg-n-times/1

class Solution:
    def printName(self, name, n):
        # Base case to terminate recursion
        if n == 0:
            return
        print(name)
        self.printName(name, n - 1)

# Example usage
solution = Solution()
solution.printName("Shreyans", 5)

